providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -312,18 +312,20 @@ static int rsa_get_params(void *key, OSSL_PARAM params[])
         return 0;
 
     /*
-     * For RSA-PSS keys, we ignore the default digest request
-     * TODO(3.0) with RSA-OAEP keys, this may need to be amended
+     * For restricted RSA-PSS keys, we ignore the default digest request.
+     * With RSA-OAEP keys, this may need to be amended.
      */
     if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_DEFAULT_DIGEST)) != NULL
-        && rsa_type != RSA_FLAG_TYPE_RSASSAPSS) {
+        && (rsa_type != RSA_FLAG_TYPE_RSASSAPSS
+            || (rsa_type == RSA_FLAG_TYPE_RSASSAPSS
+                && ossl_rsa_pss_params_30_is_unrestricted(pss_params)))) {
```

Comment:
Side note: this is an equivalent expression:

``` C
        && (rsa_type != RSA_FLAG_TYPE_RSASSAPSS
            || ossl_rsa_pss_params_30_is_unrestricted(pss_params))) {
```

Reply:
true, will amend. I want to also add a test case.

Reply:
Ah, perhaps my approval was a bit premature then.  I'll dismiss it